src/databricks/labs/ucx/hive_metastore/views_sequencer.py

```diff
@@ -89,22 +95,25 @@ def sequence(self) -> list[Table]:
         # if none, then it's safe to add that view to the next batch of views
         # the complexity for a given set of views v and a dependency depth d looks like Ov^d
         # this seems enormous but in practice d remains small and v decreases rapidly
-        table_list = self._crawler.snapshot()
         all_tables = {}
         views = set()
-        for table in table_list:
-            all_tables[table.key] = table
-            if table.view_text is None:
-                continue
-            views.add(ViewToMigrate(table))
+        tables = []
+        for table in self._tables:
+            all_tables[table.src.key] = table
+            if table.src.view_text is None:
+                tables.append(table)
+            else:
+                views.add(ViewToMigrate(table))
+        batches = [tables]
         while len(views) > 0:
             next_batch = self._next_batch(views, all_tables)
             self._result_view_list.extend(next_batch)
             self._result_tables_set.update([v.view for v in next_batch])
             views.difference_update(next_batch)
-        return [v.view for v in self._result_view_list]
+            batches.append(list(v.view for v in next_batch))
```

Comment:
```suggestion
            batches += [v.view for v in next_batch]
```
I believe below images are self-self-explantory.
<img width="1063" alt="Screenshot 2024-04-01 at 12 40 46 PM" src="https://github.com/databrickslabs/ucx/assets/3398206/135d800d-0493-48b1-a9f1-d2681bd92d44">
<img width="541" alt="Screenshot 2024-04-01 at 12 41 21 PM" src="https://github.com/databrickslabs/ucx/assets/3398206/27ebc3eb-6805-4646-865b-4b388a8569cb">